dotnet/src/IntegrationTests/Processes/ProcessTests.cs

```diff
@@ -0,0 +1,256 @@
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Processes;
+public sealed class ProcessTests
+{
+    private readonly IKernelBuilder _kernelBuilder = Kernel.CreateBuilder();
+    private readonly IConfigurationRoot _configuration = new ConfigurationBuilder()
+            .AddJsonFile(path: "testsettings.json", optional: true, reloadOnChange: true)
+            .AddJsonFile(path: "testsettings.development.json", optional: true, reloadOnChange: true)
+            .AddEnvironmentVariables()
+            .AddUserSecrets<OpenAIAssistantAgentTests>()
+            .Build();
+
+    /// <summary>
+    /// Tests a simple linear process with two steps and no sub processes.
+    /// </summary>
+    /// <returns>A <see cref="Task"/></returns>
+    [Fact]
+    public async Task LinearProcessAsync()
+    {
+        // Arrange
+        OpenAIConfiguration configuration = this._configuration.GetSection("OpenAI").Get<OpenAIConfiguration>()!;
+        this._kernelBuilder.AddOpenAIChatCompletion(
+            modelId: configuration.ModelId!,
+            apiKey: configuration.ApiKey);
+
+        Kernel kernel = this._kernelBuilder.Build();
+        var process = this.CreateLinearProcess("Simple").Build();
+
+        // Act
+        string testInput = "Test";
+        var processHandle = await process.StartAsync(kernel, new() { Id = ProcessTestsEvents.StartProcess, Data = testInput });
+        var processInfo = await processHandle.GetStateAsync();
+
+        // Assert
+        var repeatStepState = processInfo.Steps.Where(s => s.State.Name == nameof(RepeatStep)).FirstOrDefault()?.State as KernelProcessStepState<StepState>;
+        Assert.NotNull(repeatStepState?.State);
+        Assert.Equal(string.Join(" ", Enumerable.Repeat(testInput, 2)), repeatStepState.State.LastMessage);
+    }
+
+    /// <summary>
+    /// Tests a process with three steps where the third step is a nested process. Events from the outer process
+    /// are routed to the inner process.
+    /// </summary>
+    /// <returns>A <see cref="Task"/></returns>
+    [Fact]
+    public async Task NestedProcessOuterToInnerWorksAsync()
+    {
+        // Arrange
+        OpenAIConfiguration configuration = this._configuration.GetSection("OpenAI").Get<OpenAIConfiguration>()!;
+        this._kernelBuilder.AddOpenAIChatCompletion(
+            modelId: configuration.ModelId!,
+            apiKey: configuration.ApiKey);
+
+        Kernel kernel = this._kernelBuilder.Build();
+
+        // Create the outer process
+        var processBuilder = this.CreateLinearProcess("Outer");
+
+        // Create the inner process and add it as a step to the outer process
+        var nestedProcessStep = processBuilder.AddStepFromProcess(this.CreateLinearProcess("Inner"));
+
+        // Route the last step of the outer process to trigger the external event that starts the inner process
+        processBuilder.Steps[1].OnEvent(ProcessTestsEvents.OutputReadyInternal)
```

Comment:
so for chaining events the user should know about the steps within the process? 
perhaps the `GetExternalInputTriggerEvents` or similar could store the refs to steps with external outputs?